File: server/core/WebullBot.py
from webull import webull
from pprint import pprint
from core.credentials import WEBULL_DEVICE_ID, WEBULL_LOGIN_EMAIL, WEBULL_LOGIN_PWD, WEBULL_TRADING_PIN, WEBULL_DEVICE_NAME



class WebullBot:
    def __init__(self) -> None:
        self._webull = webull()
        self._loggedin = False

    
    def login(self, use_workaround: bool = False) -> bool:
        wb = self._webull
        wb._did = WEBULL_DEVICE_ID
        print(WEBULL_DEVICE_ID)
        login = wb.login(username=WEBULL_LOGIN_EMAIL, password=WEBULL_LOGIN_PWD)
        print(login)

        self._loggedin = wb.get_trade_token(WEBULL_TRADING_PIN)
        self._webull = wb

        return self._loggedin

    def positions(self):
        wb = self._webull
        return wb.get_positions()
    
    def ticker(self, ticker: str):
        wb = self._webull
        tick = wb.get_quote(ticker)
        self.print(tick['close'])
        # return {
        #     'price': float(tick['askList'][0]['price']),
        # }

    def print(self, list):
        pprint(list)

File: nodeserver/src/core/WebullBot.ts
import {
  WEBULL_DEVICE_ID,
  WEBULL_DEVICE_NAME,
  WEBULL_LOGIN_EMAIL,
  WEBULL_LOGIN_PWD,
  WEBULL_TRADING_PIN,
} from './credentials';
import { Webull } from './webull/webull';

export class WebullBot {
  async login() {
    const login = await Webull.login(
      WEBULL_LOGIN_EMAIL,
      WEBULL_LOGIN_PWD,
      WEBULL_DEVICE_NAME,
      WEBULL_DEVICE_ID,
    );
    // await python`${wb}.login(username=${WEBULL_LOGIN_EMAIL}, password=${WEBULL_LOGIN_PWD}, device_name=${WEBULL_DEVICE_ID})`;

    // this._loggedin = await python`wb.get_trade_token(WEBULL_TRADING_PIN)`;
    // this._webull = wb;
  }

  async positions() {
    // const wb = this._webull;
    // return await python`wb.get_positions()`;
  }

  async ticker(ticker) {
    const tick = await Webull.get_quote(ticker);
    return { price: Number(tick?.pPrice || tick?.close), change: tick?.change };
  }
}
